src/test/java/com/coravy/hudson/plugins/github/GithubLinkAnnotatorTest.java

```diff
@@ -25,13 +44,44 @@ public final void testAnnotateStringMarkupText() {
                         + "issues/9876/find'>close #9876</a> link");
     }
 
+    @Test
+    public final void testChangeSetAnnotateStringMarkupText() {
```

Comment:
Interesting. I diligently tried to retain the pattern and style that was already in the test, rather than switching to a Hamcrest based pattern.

I'm happy to switch to a Hamcrest based pattern and to use similar structure to XSSApiTest.

Are you willing to consider the behavioral change at the root of this request?  When the git plugin was rewritten (from git plugin 1.x to git plugin 2.x), one of the behaviors that changed was to shorten the commit messages.  I didn't implement that change in the git plugin, but I believe it is the right change.  

However, there are at least a [few git plugin 1.x users](https://issues.jenkins-ci.org/browse/JENKINS-29977) who object to the git plugin 2.x message truncation behavior.

I'm not sure if GitHub plugin users are as vocal about truncation of the SHA1, or rather if they are surprised (like I was) to see the full text of the SHA1 displayed in the web user interface.